deepchecks/nlp/checks/data_integrity/under_annotated_segments.py

```diff
@@ -0,0 +1,290 @@
+            else:  # segment by one feature
+                feature_2 = 'virtual_col'
+                feature_2_lower = encoded_data['virtual_col'].min() * 1.3
+                feature_2_upper = encoded_data['virtual_col'].max() * 1.3
+                hover_template = feature_1 + ': %{x}<br>text: %{text}<br>Annotated: '
+                tab_title = feature_1
+                msg = f'Under annotated segment contains samples with {feature_1} in range ' \
+                      f'{[format_number(feature_1_lower), format_number(feature_1_upper)]}.'
+                fig.update_yaxes(range=[feature_2_lower*1.2, feature_2_upper*1.2], tickvals=[], ticktext=[])
+
+            # Add box trace to the legend using lines
+            fig.add_trace(go.Scatter(
+                x=[feature_1_lower, feature_1_lower, feature_1_upper, feature_1_upper, feature_1_lower],
+                y=[feature_2_lower, feature_2_upper, feature_2_upper, feature_2_lower, feature_2_lower],
+                mode='lines',
+                line=dict(color='rgb(121, 100, 255)', width=3, dash='dot'),
+                name='Under Annotated Segment',
+                fill='toself', fillcolor='rgb(121, 100, 255)', opacity=0.4,
+                legendgroup='box'))
+
+            # Add annotated scatter plot
+            fig.add_trace(go.Scatter(x=annotated_data[feature_1], y=annotated_data[feature_2],
+                                     mode='markers', opacity=0.7,
+                                     marker=dict(symbol='circle', color='lightgreen', size=10,
+                                                 line=dict(color='black', width=1)),
+                                     hovertemplate=hover_template + 'True',
+                                     text=annotated_data['text'],
+                                     name='Annotated Samples'))
+
+            # Add not annotated scatter plot
+            fig.add_trace(go.Scatter(x=not_annotated_data[feature_1], y=not_annotated_data[feature_2],
+                                     mode='markers', opacity=0.7,
+                                     marker=dict(symbol='x', color='red', size=10, line=dict(color='black', width=1)),
+                                     hovertemplate=hover_template + 'False',
+                                     text=not_annotated_data['text'],
+                                     name='Not Annotated Samples'))
+
+            # Update figure layout
+            fig.update_layout(title=dict(
+                text=f'Under Annotated Segment ({row["% of Data"]}% of Data)<br><sub>Annotation ratio in segment ' \
+                     f'is {format_percent(row["Annotation Ratio"])} (vs. {format_percent(avg_score)}'
+                     f' in whole data)</sub>',
+                font=dict(size=24)),
+                xaxis_title=feature_1, yaxis_title=feature_2 if feature_2 != 'virtual_col' else '',
+                autosize=False, width=1000, height=600,
```

Comment:
If using constant size test on various displays (big screen, mac screen, 13'' screen (nir has one))

Reply:
That how it was defined in our plot files - see confusion matrix abstract, core/display.py
Anyhow could make it work with aspect ratio after spending 30-min and its not worth more